### DISCUSSION TITLE
unable to build from source

### DISCUSSION THREAD

**Discussion by MetalDude101** [likec4/likec4#633] [Q&A] [answered]
I am new to react based apps and unfamiliar with the tools.
Is there a place that mentions the exact tools, steps, commands (in order) to build likec4?
I went down a rabbit hole of: npm, pnpm, turbo?, yarn, etc...
Also, your elements in a "workspace" seem non-standard by one of the tools.
I need some help to do this, so i can show how this works to management.
Thanks

**Comment by davydkov** [comment, marked as answer]
Hi @MetalDude101 (nice nickname)

Do I understand you right, you want to build likec4 from sources, this repository?
If so, take a look at GitHub workflows.
It always helps me to understand how projects are built, tested, etc. 

As for LikeC4, you need:
- nodejs 20.x
- yarn cli (like `1.22.21`), that delegates to the version at `.yarn/releases`

Then, in root:
```bash
yarn install
yarn build
# or better
yarn typecheck 
```

`typecheck` triggers `build`, but also gets caches of ts compile results.


#### Development workflow

**Option 1:** there is [launch configuration](https://github.com/likec4/likec4/blob/main/.vscode/launch.json#L19-L41) to run/debug extension.
Hint: I also keep running `yarn dev` in [packages/vscode-preview](https://github.com/likec4/likec4/blob/main/packages/vscode-preview).
This rebuilds Preview panel. If I change only the UI part, it's enough to close and open the panel in a running extension (no need to restart it).

**Option 2:** my main workflow to work on anything else except vscode extension.
Run `yarn dev` in [packages/likec4](https://github.com/likec4/likec4/blob/main/packages/likec4) (there are multiple "dev" commands to use different .c4 sources).

And sometimes, keep running `yarn vitest:ui` in the root.


> Also, your elements in a "workspace" seem non-standard by one of the tools.

Yes, that's right.
Package exports ts-files, but when published, exports js files.
But this is only to make development more comfortable (like no need to run watch tasks).

**Reply by MetalDude101** [reply]
What is the expected output of all this? This page: https://likec4.dev/docs/tools/cli/  mentions lots of tools. It looks like, at some point, an executable is created. "Then, you can call likec4 directly:  likec4 [command] "
How can i get a website created to host\run likec4?
